src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_WORK_INDEX = 2;
+    private static final int SALARY_INDEX = 3;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+
+        StringBuilder resultBuilder = new StringBuilder();
+        resultBuilder.append("Report for period ").append(dateFrom).append(" - ")
+                .append(dateTo);
+        Date fromDate = null;
+        Date toDate = null;
+        try {
+            fromDate = DATE_FORMAT.parse(dateFrom);
+            toDate = DATE_FORMAT.parse(dateTo);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
wrapping exception changes nothing
```suggestion
        Date fromDate = DATE_FORMAT.parse(dateFrom);
        Date toDate = DATE_FORMAT.parse(dateTo);
```